torch/distributed/tensor/_utils.py

```diff
@@ -246,6 +246,60 @@ def compute_global_tensor_info(
     return tensor_shape, tensor_stride
 
 
+def compute_global_tensor_shape(
+    shape: torch.Size, mesh: DeviceMesh, placements: Sequence[Placement]
+) -> torch.Size:
+    """
+    Compute the global size of a DTensor from the given local tensor shape,
+    the mesh and placements. Different from `compute_global_tensor_info`,
+    which assumes sharding is even, this util allgathers local shards' shapes
+    from all ranks and thus can support uneven sharding.
+    NOTE: Currently this function only supports 1D mesh.
+
+    Args:
+        shape (:class:`torch.Size`):
+            Shape of the local tensor
+        mesh (:class:`DeviceMesh`):
+            Object which describes the mesh topology
+            of devices for the DTensor.
+        placements (Sequence[:class:`Placement`]]):
+            The attribute of the DTensor that describes its layout
+            on the mesh topology.
+
+    Return:
+        tensor_shape: Shape of the global DTensor.
+    """
+    if mesh.ndim > 1:
+        raise NotImplementedError(
+            "compute_global_tensor_shape only supports 1D mesh for now."
+        )
+    if len(placements) != 1:
+        raise NotImplementedError(
+            "compute_global_tensor_shape only supports 1 placement for now."
+        )
+
+    if isinstance(placements[0], Replicate):
+        return shape
+    elif isinstance(placements[0], Shard):
+        local_shape = torch.tensor(list(shape))
+        gathered_shaped_tensors = [
+            torch.empty_like(local_shape, device=local_shape.device)
+            for _ in range(torch.distributed.get_world_size())
```

Comment:
`torch.distributed.get_world_size()` -> `self.world_size`

Reply:
Here we would not have a `self` object, may be can use `mesh.size()` ?

Reply:
Right. This is the right value to use. Think about a world that's separated into multiple parallelism axes (e.g. FSDP + TP), then `WORLD_SIZE` != `mesh.size()` if DTensor is sharded over 1D mesh (e.g. TP).